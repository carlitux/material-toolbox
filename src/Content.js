// @flow
import * as React from 'react';
import { Route } from 'react-router-dom';  // eslint-disable-line

import LayoutGrid from './material-toolbox/layout-grid/Grid';
import LayoutInner from './material-toolbox/layout-grid/Inner';

import Hero from './Hero';
import HomePage from './HomePage';

import AnimationPage from './AnimationPage';
import CheckboxesPage from './CheckboxesPage';
import FormFieldsPage from './FormFieldsPage';
import RipplesPage from './RipplesPage';
import SnackbarPage from './SnackbarPage';
import TabsPage from './TabsPage';
import TextFieldPage from './TextFieldPage';

const Content = () => (
  <React.Fragment>
    <Route exact path="/" component={Hero} />
    <LayoutGrid>
      <LayoutInner>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/animation" component={AnimationPage} />
        <Route
          exact
          path="/input-controls/checkboxes"
          component={CheckboxesPage}
        />
        <Route
          exact
          path="/input-controls/form-fields"
          component={FormFieldsPage}
        />
        <Route
          exact
          path="/input-controls/text-fields"
          component={TextFieldPage}
        />
        <Route exact path="/ripples" component={RipplesPage} />
        <Route exact path="/snackbar" component={SnackbarPage} />
        <Route exact path="/tabs" component={TabsPage} />
      </LayoutInner>
    </LayoutGrid>
  </React.Fragment>
);

export default Content;
