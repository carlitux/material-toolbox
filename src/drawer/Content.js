// @flow
import * as React from 'react';
import classnames from 'classnames';

type Props = {
  children: React.Node,
  className: string,
  drawerType: 'temporary' | 'persistent' | 'permanent',
};

const DrawerContent = ({ drawerType, children, className, ...rest }: Props) => {
  const composedClassName = classnames(
    className,
    `mdc-${drawerType}-drawer__content`,
  );

  return React.Children.map(children, child => (
    <child.type {...child.props} className={composedClassName} />
  ));
};

export default DrawerContent;
