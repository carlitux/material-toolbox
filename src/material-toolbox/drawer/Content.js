// @flow
import * as React from 'react';
import classnames from 'classnames';

type Props = {
  children: React.Node,
  className?: string,
};

const DrawerContent = ({ children, className }: Props) => {
  const composedClassName = classnames(className, 'mdc-drawer__content');

  return React.Children.map(children, child => (
    <child.type {...child.props} className={composedClassName} />
  ));
};

export default DrawerContent;
