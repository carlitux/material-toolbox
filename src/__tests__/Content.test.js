import React from 'react';
import { shallow } from 'enzyme';
import Content from '../Content';

describe('component::Content', () => {
  it('Should match snapshoot', () => {
    const component = shallow(<Content />);
    expect(component).toMatchSnapshot();
  });
});
