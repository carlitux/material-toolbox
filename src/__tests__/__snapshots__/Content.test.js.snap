// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::Content Should match snapshoot 1`] = `
<React.Fragment>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Grid
    fixed={false}
  >
    <Inner>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/animation"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/input-controls/checkboxes"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/input-controls/form-fields"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/input-controls/text-fields"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/ripples"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/snackbar"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/tabs"
      />
    </Inner>
  </Grid>
</React.Fragment>
`;
